src/device/usbd.c

```diff
@@ -287,19 +290,19 @@ void tud_task (void)
         {
           // Invoke the class callback associated with the endpoint address
           uint8_t const ep_addr = event.xfer_complete.ep_addr;
-          uint8_t const epnum = tu_edpt_number(ep_addr);
-          uint8_t const dir   = tu_edpt_dir(ep_addr);
+          uint8_t const epnum   = tu_edpt_number(ep_addr);
+          uint8_t const ep_dir  = tu_edpt_dir(ep_addr);
 
-          _usbd_dev.ep_busy_map[dir] = (uint8_t) tu_bit_clear(_usbd_dev.ep_busy_map[dir], epnum);
+          _usbd_dev.ep_busy_map[ep_dir] = (uint8_t) tu_bit_clear(_usbd_dev.ep_busy_map[ep_dir], epnum);
 
-          if ( 0 == tu_edpt_number(ep_addr) )
+          if ( 0 == epnum )

```

Comment:
Do we need to call the USBD control_xfer_fb  CLEAR _FEATURE, GET_STATUS, or SET_FEATURE that we need to handle for EP other than 0? (so, call the local one if the driver didn't handle the request?) If both we and the driver handle a request, who gets to set the status?

Reply:
All Control requests are alway sent to ep0. The setup packet data contains the targeted endpoint

Reply:
My mistake. I was confused with the wIndex/target ep=0, not the ep over which the packet came.